circuitpython/network/golemio_odjezdy_mhd/code.py

```diff
@@ -0,0 +1,132 @@
+"""
+What about using some open data that are provided by your government or city?
+In this example we will connect to Prague data platform Golemio.cz and display departure times at some public transport stop.
+
+1. To get started, let's set up an API Golemio account, and generate your API key for free at https://api.golemio.cz/api-keys/auth/sign-up Don't forget to set the api key as an environment variable  GOLEMIO_API_KEY in the settings.toml file.
+2. Next, set the WiFi SSID and password as environment variables CIRCUITPY_WIFI_SSID and CIRCUITPY_WIFI_PASSWORD in the settings.toml file.
+3. Finally set stop STOP_NAME or STOP_GTFS_ID in settings.toml file for stop that you want you inspect.
+   - STOP_NAME is case-sensitive fullname of stop, MUST BE written with diacritics, no acronyms are accepted.
+   - GTFS ID (General Transit Feed Specification ID) is unique id for every stop, where you can be more selective,
+     eg. U400Z101P is GTFS ID for Muzeum metro A direction Depo Hostivar.
+     List of GTFS ID can be found at https://data.pid.cz/stops/json/stops.json
+
+If you are interested than you might check departure times at https://mapa.pid.cz/ they should be exatly same like in this script.
+
+You can find the required libraries in the CircuitPython library bundle (https://circuitpython.org/libraries).
+"""
+import os
+import wifi
+import ssl
+import socketpool
+import time
+import zlib
+import json
+import adafruit_requests
+
+
+# If you have font with Czech diacritics than you will not need this
+def removeCzechDiacritics(text):
```

Comment:
Your current function for removing Czech diacritics is limited and inefficient. It doesn't handle German characters like ö in "Högerova" or ü in "Klando,Kübeck", which appear in some stop names. Instead of reinventing the wheel, I recommend using Python's built-in unicodedata module.

```python
import unicodedata

def remove_diacritics(text):
    return ''.join(c for c in unicodedata.normalize('NFKD', text)
                   if not unicodedata.combining(c))
```

Reply:
![image](https://github.com/user-attachments/assets/4b5c1eb2-f5f7-467c-80cb-25cd66099c99)

@asciidan-oict  I don't believe such module exist in [Circuitpython]
(https://docs.circuitpython.org/en/latest/README.html).

And I have not found any relevant module in adafruit-circuitpython space.

![image](https://github.com/user-attachments/assets/cace5aa6-8d7d-46b0-998f-7cf5467b3c97)

I will expand this funcion for German characters. It should be there only temporarily anyway, until @lynt-smitka finish Czech localization of Circuitpython. After that this function should be irelevant.

Reply:
I plan to go through the code and add a mini font trick to work with accents (I think it doesn't count German characters, but it's not a problem to add them). I think this is the most elegant way to display these characters. https://github.com/MakerClassCZ/Events/tree/main/random_help/czech_font

Reply:
@lynt-smitka Will such custom font affect just functions with adafruit_display_text or also screen font used in print to STRDOUT (that is used in this example)?

Reply:
@asciidan-oict I have checked all dangerous characters in stop_names, trop_headsigns and routes_short_names. The actual list is `ÁÉĚÍÓÖÚŮÝČĎŇŘŠŤŽáéěíóöúůüýčďňřšťž✈` I belive now is is robuts enough.

Bash script for obtaining all characters:
```
GTFS_ZIP_FILEMAME='pid_gtfs.zip'
GTFS_DIRNAME='pid_gtfs'

# check for file GTFS_ZIP_FILEMAME
 if [[ ! -e "$GTFS_ZIP_FILEMAME" ]]; then
	curl "https://data.pid.cz/PID_GTFS.zip" --output "$GTFS_ZIP_FILEMAME"
 fi

# check for directory GTFS_DIRNAME
if [[ ! -d "$GTFS_DIRNAME" ]]; then 
	unzip pid_gtfs.zip -d "$GTFS_DIRNAME"
fi

duckdb -csv -c "SELECT stop_name FROM read_csv_auto('pid_gtfs/stops.txt')" > all_stop_names.txt
grep -v "^stop_name$" all_stop_names.txt | grep -o . | sort | uniq > character_in_stop_names.txt
# "&()+,-./0123456789=ABCDEFGHIJKLMNOPRSTUVWXYZabcdefghijklmnopqrstuvwxyzÁÚáéíóöúüýČčĎďěňŘřŠšťůŽž

duckdb -csv  -c "SELECT trip_headsign FROM read_csv_auto('pid_gtfs/trips.txt')" > all_trip_headsigns.txt
grep -v "^trip_headsign$" all_trip_headsigns.txt | grep -o . | sort | uniq > character_in_trip_headsigns.txt
# "&(),-./1ABCDEFGHIJKLMNOPRSTUVWYZabcdefghijklmnopqrstuvxyzÁÖÚáéíóúüýČčĎďěňŘřŠšťůŽž✈

duckdb -csv  -c "SELECT route_short_name FROM read_csv_auto('pid_gtfs/routes.txt')" > all_routes_short_names.txt
grep -v "^route_short_name$" all_routes_short_names.txt | grep -o . | sort | uniq > character_in_routes_short_names.txt
# "0123456789ABCDEHKLMNOPRSTUVXYceilrÍŠ
```